File: week-05/day-05/garden-again/tree.ts
import { Plant } from "./plant";

'use strict';

export class Tree extends Plant {

  constructor(color: string){
    super(color)
    this.absorbingCap = 0.4;
  }

  needWater(): boolean {
    return this.currentWaterAm<10;
  }  
  
  getType() {
    return 'tree'
  }
  
}

File: datascience/week-01/garden/garden_py/tree.py
import random
import sys
import os
from plant import Plants


class Tree(Plants):

    def __init__(self, color):
        super().__init__(color)
        self._absorbingCap = 0.4

    def needWater(self):
        return self._currentWaterAm < 10

    def get_type(self):
        return 'tree'
